### ISSUE TITLE
Code referencing longdouble array will not decompile

### ISSUE THREAD

**Issue by jbw3** [NationalSecurityAgency/ghidra#5762]
**Describe the bug**

Ghidra fails when trying to decompile a function that references a long double array.

**To Reproduce**

1. Compile the following C program with GCC:
```c
#include <stdio.h>

int main()
{
    long double x[2] = {1.23, 4.56};
    printf("%Lf %Lf\n", x[0], x[1]);

    return 0;
}
```

GCC command:
```
gcc -g main.c -o bug
```

2. Open the program in Ghidra and go to `main`
3. You will see the following error: "Low-level Error: Bad size for array of type longdouble"

**Environment (please complete the following information):**
 - OS: Ubuntu 22.04
 - Java Version: 17.0.2
 - Ghidra Version: 10.3.3
 - Ghidra Origin: official GitHub distro

**Comment by LukeSerne** [comment]
This is an issue in the decompiler, specifically in [`TypeArray::decode`](https://github.com/NationalSecurityAgency/ghidra/blob/77373649fd96bc3f7c6c34bc76df0b276d913131/Ghidra/Features/Decompiler/src/decompile/cpp/type.cc#L1136-L1156). The reason the error is thrown is because `arraysize * arrayof->getSize() != size` - the array is `longdouble[2]`, where `longdouble` refers to an 80-bit float, padded to 128 bits. So, `arraysize` is `2`, `arrayof->getSize()` is `10` (each element is 80 bits = 10 bytes long), and `size` is `32` (the total size of the array, with padding of the elements). Clearly, `2 * 10 == 20 != 32`. This check assumes that all elements of the array are "tightly packed" (there is no padding between elements), but there is padding here, so the check fails.

You can work around this issue by editing the stack of the problematic function (right-click the function name, `Function > Edit Stack Frame`) and changing the type of the array to something else. Once the `longdouble` array is no longer on the stack, the decomiler will decompile the function. If the decompiler can't optimise the stack variable away, the decompiled output will suffer though.

Note that you can create a `longdouble` array in the listing view just fine. It will interpret and display the constants correctly.

Since this issue is specific to Ghidra's implementation of the `longdouble` type and requires the stack variable to have the type `longdouble[2]` (which is set by parsing the DWARF debugging info), it only occurs for binaries that are compiled for platforms where `gcc` compiles `long double` to an 80-bit extended floating point type and with DWARF debugging info from a version less than 5 (because of #4088). To help further analysing this issue, I have attached the compiled binary (Compiled with `gcc 13.2.1` using `gcc -gdwarf-4 main.c -o bug4`), and the decompiler debug xml in [this zip](https://github.com/NationalSecurityAgency/ghidra/files/12611351/issue_5762.zip).

**Comment by dev747368** [comment]
Thanks for the main.c example.  The root cause is as you mentioned, the discrepancy between the sizeof and alignedsize of those longdoubles, which is a new concept that is still getting the kinks worked out.

We will update this ticket when there is some progress.

**Comment by caheckman** [comment]
Addressed by commit ed4604fde10683627943d704e9aa0b2306415bcf
